cmd/mesh/manifest-migrate.go

```diff
@@ -15,23 +15,131 @@
+)
+
+const (
+	defaultNamespace = "istio-system"
 )
 
-func manifestMigrateCmd(rootArgs *rootArgs) *cobra.Command {
+type manifestMigrateArgs struct {
+	// migrateDir indicates whether to migrate all values.yaml files in directory
+	migrateDir bool
+	// namespace is the namespace to get the in cluster configMap
+	namespace string
+}
+
+func addManifestMigrateFlags(cmd *cobra.Command, args *manifestMigrateArgs) {
+	cmd.PersistentFlags().BoolVarP(&args.migrateDir, "migrateDir", "r", false, "translate directory")
+	cmd.PersistentFlags().StringVarP(&args.namespace, "namespace", "n", defaultNamespace, "specifies namespace")
+}
+
+func manifestMigrateCmd(rootArgs *rootArgs, mmArgs *manifestMigrateArgs) *cobra.Command {
 	return &cobra.Command{
 		Use:   "migrate",
 		Short: "Migrates a file containing Helm values to IstioControlPlane format.",
 		Long:  "The migrate subcommand is used to migrate a configuration in Helm values format to IstioControlPlane format.",
-		Args:  cobra.ExactArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
-			manifestMigrate(rootArgs)
+			if len(args) == 0 {
+				migrateFromClusterConfig(rootArgs, mmArgs)
+			} else {
+				migrateFromFiles(rootArgs, args, mmArgs)
+			}
 		}}
+}
 
+func valueFileFilter(path string) bool {
+	return filepath.Base(path) == "values.yaml"
 }
 
-func manifestMigrate(_ *rootArgs) {
-	fmt.Println("Not yet implemented.")
+// migrateFromFiles handles migration for local values.yaml files
+func migrateFromFiles(rootArgs *rootArgs, args []string, mmArgs *manifestMigrateArgs) {
+	if err := configLogs(rootArgs); err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Could not configure logs: %s", err)
+		os.Exit(1)
+	}
+	logAndPrintf(rootArgs, "translating input values.yaml file at: %s to new API", args[0])
+	if mmArgs.migrateDir {
+		value, err := util.ReadFromDir(args[0], valueFileFilter)
+		if err != nil {
+			logAndFatalf(rootArgs, err.Error())
+		}
+		translateFunc(rootArgs, []byte(value))
+	} else {
+		value, err := ioutil.ReadFile(args[0])
+		if err != nil {
+			logAndFatalf(rootArgs, err.Error())
+		}
+		translateFunc(rootArgs, value)
+	}
+}
+
+// translateFunc translates the input values and output the result
+func translateFunc(rootArgs *rootArgs, values []byte) {
+	ts, err := translate.NewValueYAMLTranslator(version.NewMinorVersion(1, 3))
+	if err != nil {
+		logAndFatalf(rootArgs, "error creating values.yaml translator: %s", err.Error())
+	}
+
+	valueStruct := v1alpha2.Values{}
+	err = yaml.Unmarshal(values, &valueStruct)
+	if err != nil {
+		logAndFatalf(rootArgs, "error unmarshalling values.yaml into value struct : %s", err.Error())
+	}
+
+	isCPSpec, err := ts.TranslateFromValueToSpec(&valueStruct)
+	if err != nil {
+		logAndFatalf(rootArgs, "error translating values.yaml: %s", err.Error())
+	}
+	isCPYaml, err := yaml.Marshal(isCPSpec)
+	if err != nil {
+		logAndFatalf(rootArgs, "error marshalling translated IstioControlPlaneSpec: %s", err.Error())
+	}
+
+	fmt.Println(string(isCPYaml))
```

Comment:
This should be a writer in case -o is set.

Reply:
the -o flag was actually removed based on last review

Reply:
ah, you're right - sorry.